hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/source/IncrementalInputSplits.java

```diff
@@ -601,9 +602,9 @@ public List<HoodieInstant> filterInstantsWithRange(
    * @return the filtered timeline
    */
   @VisibleForTesting
-  public HoodieTimeline filterInstantsAsPerUserConfigs(HoodieTimeline timeline) {
+  public HoodieTimeline filterInstantsAsPerUserConfigs(HoodieTimeline timeline, HoodieTableType tableType) {
     final HoodieTimeline oriTimeline = timeline;
-    if (this.skipCompaction) {
+    if (tableType.equals(HoodieTableType.MERGE_ON_READ) & this.skipCompaction) {
```

Comment:
Another choice is to resolve the table type with `OptionsResolver` and the `this.conf`.

Reply:
Adjusted to use OptionsResolver.